pkg/k8s/testutils/decoder.go

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package testutils
+
+import (
+	"os"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+
+	cilium_v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+	slim_fake "github.com/cilium/cilium/pkg/k8s/slim/k8s/client/clientset/versioned/fake"
+)
+
+// Decoder is an object decoder for Cilium and Slim objects.
+// The [DecodeObject] and [DecodeFile] functions are provided as
+// shorthands for decoding from bytes and files respectively.
+var Decoder runtime.Decoder
+
+func init() {
+	scheme := runtime.NewScheme()
+	slim_fake.AddToScheme(scheme)
+	cilium_v2.AddToScheme(scheme)
+	Decoder = serializer.NewCodecFactory(scheme).UniversalDeserializer()
```

Comment:
Do you think it would be useful to have a way to add to this scheme later? It will be difficult to use this for Gateway API objects, for example, in the future without a way to add more to the scheme.

I'm not sure how that would interact with other StateDB assumptions though, interested in your thoughts.

Reply:
This isn't related to StateDB in any way (other than that I use it in `pkg/k8s/statedb_test.go` to test it). Definitely the idea is to have everything we use in this scheme. I'll rework it to add everything we use in OSS and also make it extendable from forked codebases.